pkg/endpoints/endpoints_test.go

```diff
@@ -0,0 +1,146 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoints
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestParseFlags(t *testing.T) {
+	testCases := []struct {
+		name           string
+		flagToParse    string
+		expectedOutput []ServiceEndpoint
+		expectedError  error
+	}{
+		{
+			name:           "no configuration",
+			flagToParse:    "",
+			expectedOutput: nil,
+			expectedError:  nil,
+		},
+		{
+			name:        "single region, single vpc service",
+			flagToParse: "eu-gb:vpc=https://vpchost:8080",
+			expectedOutput: []ServiceEndpoint{
+				{
+					ID:     "vpc",
+					URL:    "https://vpchost:8080",
+					Region: "eu-gb",
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:        "single region, single powervs service",
+			flagToParse: "lon:powervs=https://pvshost:8080",
+			expectedOutput: []ServiceEndpoint{
+				{
+					ID:     "powervs",
+					URL:    "https://pvshost:8080",
+					Region: "lon",
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:        "single region, multiple services",
+			flagToParse: "lon:powervs=https://pvshost:8080,rc=https://rchost:8080",
+			expectedOutput: []ServiceEndpoint{
+				{
+					ID:     "powervs",
+					URL:    "https://pvshost:8080",
+					Region: "lon",
+				},
+				{
+					ID:     "rc",
+					URL:    "https://rchost:8080",
+					Region: "lon",
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:           "single region, duplicate service",
+			flagToParse:    "eu-gb:vpc=https://localhost:8080,vpc=https://vpchost:8080",
+			expectedOutput: nil,
+			expectedError:  errServiceEndpointDuplicateID,
+		},
+		{
+			name:           "single region, non-valid URI",
+			flagToParse:    "eu-gb:vpc=fdsfs",
+			expectedOutput: nil,
+			expectedError:  errServiceEndpointURL,
+		},
+		{
+			name:        "multiples regions",
+			flagToParse: "eu-gb:vpc=https://vpchost:8080;lon:powervs=https://pvshost:8080,rc=https://rchost:8080",
+			expectedOutput: []ServiceEndpoint{
+				{
+					ID:     "vpc",
+					URL:    "https://vpchost:8080",
+					Region: "eu-gb",
+				},
+				{
+					ID:     "powervs",
+					URL:    "https://pvshost:8080",
+					Region: "lon",
+				},
+				{
+					ID:     "rc",
+					URL:    "https://rchost:8080",
+					Region: "lon",
+				},
+			},
+			expectedError: nil,
+		},
+		{
+			name:           "invalid config",
+			flagToParse:    "eu-gb=localhost",
+			expectedOutput: nil,
+			expectedError:  errServiceEndpointRegion,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			out, err := ParseServiceEndpointFlag(tc.flagToParse)
+
+			if !errors.Is(err, tc.expectedError) {
+				t.Fatalf("did not expect correct error: got %v, expected %v", err, tc.expectedError)
+			}
+
+			if !endpointsEqual(out, tc.expectedOutput) {
```

Comment:
see if you can use `assert.EqualSlice` from github.com/stretchr/testify/assert package

Reply:
`EqualSlice` is replaced with `ElementsMatch` in assert package using that.